frontend/src/app/testQueue/TestCardForm/diseaseSpecificComponents/aoeUtils.ts

```diff
@@ -0,0 +1,219 @@
+  };
+}
+
+export function generateSexualActivityAoeConstants(
+  responses: AoeQuestionResponses,
+  hasAttemptedSubmit: boolean
+) {
+  const isPregnancyFilled = !!responses.pregnancy;
+  const isGenderOfSexualPartnersAnswered =
+    !!responses.genderOfSexualPartners &&
+    responses.genderOfSexualPartners.length > 0;
+
+  const showPregnancyError = hasAttemptedSubmit && !isPregnancyFilled;
+  const showGenderOfSexualPartnersError =
+    hasAttemptedSubmit && !isGenderOfSexualPartnersAnswered;
+
+  const isSyphilisHistoryFilled = !!responses.syphilisHistory;
+  const showSyphilisHistoryError =
+    hasAttemptedSubmit && !isSyphilisHistoryFilled;
+
+  const selectedGenders: string[] = [];
+  responses.genderOfSexualPartners?.forEach((g) => {
+    if (g) {
+      selectedGenders.push(g);
+    }
+  });
+
+  return {
+    showPregnancyError,
+    showGenderOfSexualPartnersError,
+    showSyphilisHistoryError,
+    selectedGenders,
+  };
+}
+
+export function generateSymptomAoeConstants(
+  responses: AoeQuestionResponses,
+  hasAttemptedSubmit: boolean,
+  symptomDefinitionMap: SymptomDefinitionMap[]
+) {
+  const isSymptomsAnswered = typeof responses.noSymptoms == "boolean";
+  const isSymptomOnsetDateAnswered = !!responses.symptomOnset;
+
+  const showSymptomsError = hasAttemptedSubmit && !isSymptomsAnswered;
+  const showSymptomOnsetDateError =
+    hasAttemptedSubmit &&
+    isSymptomsAnswered &&
+    responses.noSymptoms === false &&
+    !isSymptomOnsetDateAnswered;
+
+  let hasSymptoms: YesNo | undefined = undefined;
+  if (responses.noSymptoms) {
+    hasSymptoms = "NO";
+  }
+  if (responses.noSymptoms === false) {
+    hasSymptoms = "YES";
+  }
+  const symptoms: Record<string, boolean> = mapSymptomBoolLiteralsToBool(
+    responses.symptoms,
+    symptomDefinitionMap
+  );
+
+  const isSymptomOnsetAnswered =
+    !!responses.symptoms && Object.values(symptoms).some((x) => x?.valueOf());
+
+  const showSymptomOnsetError =
+    hasAttemptedSubmit &&
+    isSymptomsAnswered &&
+    responses.noSymptoms === false &&
+    !isSymptomOnsetAnswered;
+
+  return {
+    showSymptomsError,
+    showSymptomOnsetDateError,
+    hasSymptoms,
+    showSymptomOnsetError,
+    symptoms,
+  };
+}
+
+export function stringifySymptomJsonForAoeUpdate(
+  symptomJson: string | undefined | null
+) {
+  let symptomPayload = "{}";
+  if (!symptomJson) return symptomPayload;
+
+  const parsedSymptoms = JSON.parse(symptomJson);
+  const symptomKeys = Object.keys(parsedSymptoms);
+
+  if (_.isEqual(symptomKeys, Object.keys(respiratorySymptomsMap))) {
+    symptomPayload = JSON.stringify(
+      mapSymptomBoolLiteralsToBool(symptomJson, respiratorySymptomDefinitions)
+    );
+  }
+  if (_.isEqual(symptomKeys, Object.keys(syphilisSymptomsMap))) {
+    symptomPayload = JSON.stringify(
+      mapSymptomBoolLiteralsToBool(symptomJson, syphilisSymptomDefinitions)
+    );
+  }
+  return symptomPayload;
+}
+
+export function mapSpecifiedSymptomBoolLiteralsToBool(
```

Comment:
Currently I'm not seeing this be used anywhere -- will this be leveraged once Syphilis is introduced?

Reply:
yep! realized I didn't carve it out in the selective rebase but it'll be used in the followup